x/emissions/keeper/queryserver/query_server_inferences.go

```diff
@@ -164,17 +185,131 @@ func (qs queryServer) GetLatestAvailableNetworkInference(
 			inferenceBlockHeight,
 			previousLossBlockHeight,
 		)
-
 	if err != nil {
 		return nil, err
 	}
 
+	ciRawPercentiles, ciValues, err :=
+		qs.GetConfidenceIntervalsForInferenceData(
+			networkInferences,
+			forecastImpliedInferenceByWorker,
+			infererWeights,
+			forecasterWeights,
+		)
+
+	if ciRawPercentiles == nil {
+		ciRawPercentiles = []alloraMath.Dec{}
+	}
+
+	if ciValues == nil {
+		ciValues = []alloraMath.Dec{}
+	}
+
 	return &types.QueryLatestNetworkInferencesResponse{
-		NetworkInferences:         networkInferences,
-		InfererWeights:            synth.ConvertWeightsToArrays(infererWeights),
-		ForecasterWeights:         synth.ConvertWeightsToArrays(forecasterWeights),
-		ForecastImpliedInferences: synth.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
-		InferenceBlockHeight:      inferenceBlockHeight,
-		LossBlockHeight:           previousLossBlockHeight,
+		NetworkInferences:                networkInferences,
+		InfererWeights:                   synth.ConvertWeightsToArrays(infererWeights),
+		ForecasterWeights:                synth.ConvertWeightsToArrays(forecasterWeights),
+		ForecastImpliedInferences:        synth.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
+		InferenceBlockHeight:             inferenceBlockHeight,
+		LossBlockHeight:                  previousLossBlockHeight,
+		ConfidenceIntervalRawPercentiles: ciRawPercentiles,
+		ConfidenceIntervalValues:         ciValues,
 	}, nil
 }
+
+func (qs queryServer) GetConfidenceIntervalsForInferenceData(
+	networkInferences *emissions.ValueBundle,
+	forecastImpliedInferenceByWorker map[string]*emissions.Inference,
+	infererWeights map[string]alloraMath.Dec,
+	forecasterWeights map[string]alloraMath.Dec,
+) ([]alloraMath.Dec, []alloraMath.Dec, error) {
+
+	inferersToInference := make(map[string]float64)
+	for _, infererValue := range networkInferences.InfererValues {
+		floatValue, err := stringToFloat64(infererValue.Value.String())
+		if err != nil {
+			return nil, nil, err
+		}
+		inferersToInference[infererValue.Worker] = floatValue
+	}
+
+	forecastersToInference := make(map[string]float64)
+	for _, forecasterValue := range networkInferences.ForecasterValues {
+		floatValue, err := stringToFloat64(forecasterValue.Value.String())
+		if err != nil {
+			return nil, nil, err
+		}
+		forecastersToInference[forecasterValue.Worker] = floatValue
+	}
+
+	inferersToWeights := make(map[string]float64)
+	for worker, weight := range infererWeights {
+		floatValue, err := stringToFloat64(weight.String())
+		if err != nil {
+			return nil, nil, err
+		}
+		inferersToWeights[worker] = floatValue
+	}
+
+	forecastersToWeights := make(map[string]float64)
+	for worker, weight := range forecasterWeights {
+		floatValue, err := stringToFloat64(weight.String())
+		if err != nil {
+			return nil, nil, err
+		}
+		forecastersToWeights[worker] = floatValue
+	}
```

Comment:
Why we are using `float64` here instead of `Dec`?

Reply:
agreed we should not have floats anywhere in the codebase, they break consensus because CPU architectures implement finer details of float IEEE standard differently